lib/menu-items-batch-processing/class-operation.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace WP_REST_Menu_Items_Batch_Operation;
+
+use \WP_Error;
+use \WP_REST_Menu_Items_Controller;
+
+abstract class Operation {
+	const INSERT = 'insert';
+	const UPDATE = 'update';
+	const DELETE = 'delete';
+
+	protected $controller;
+	public $input;
+	/** @var Operation */
+	public $parent;
+	public $prepared_item;
+	public $result;
+
+	/**
+	 * Operation constructor.
+	 *
+	 * I really dislike the dependency on $controller, but core code is quite interlocked. I don't think
```

Comment:
Yes I can only +1 this. This dependency also makes the `Batch_Processor` feel less like an utility.